core/controllers/base_test.py

```diff
@@ -1423,3 +1469,397 @@ def test_that_correct_derived_class_does_not_raise_exception(self):
+                self.log_line(
+                    'Missing keys in HANDLER_ARGS_SCHEMAS for %s: %s.' % (
+                        handler_class_name, ', '.join(missing_schema_keys)))
+
+        error_msg = (
+            'Missing schema keys in HANDLER_ARGS_SCHEMAS for [ %s ] classes.'
+            '\nVisit %s to know, how to write schema for handler args.' % (
+                ', '.join(handlers_with_missing_request_schema_keys),
+                    self.wiki_page_link))
+
+        self.assertEqual(
+            handlers_with_missing_request_schema_keys, [], error_msg)
+
+    def test_default_value_in_schema_conforms_with_schema(self):
+        """This test check whether the default_value provided in schema
+        conforms with the schema.
+        """
+        handlers_with_non_conforming_default_schemas = []
+        list_of_routes_which_needs_schema = (
+            self._get_list_of_routes_which_need_schemas())
+
+        for route in list_of_routes_which_needs_schema:
+            handler = route.handler
+
+            handler_class_name = handler.__name__
+            if handler_class_name in self.handler_class_names_with_no_schema:
+                continue
+            if handler.HANDLER_ARGS_SCHEMAS is None:
+                continue
+
+            schemas = handler.HANDLER_ARGS_SCHEMAS
+            for request_method, request_method_schema in schemas.items():
+                for arg, schema in request_method_schema.items():
+                    if 'default_value' not in schema:
+                        continue
+                    default_value = {arg: schema['default_value']}
+                    default_value_schema = {arg: schema}
+
+                    errors = payload_validator.validate(
+                        default_value, default_value_schema, True)
+                    if len(errors) == 0:
+                        continue
+                    self.log_line(
+                        'Handler: %s, argument: %s, default_value '
+                            'validation failed.' % (handler_class_name, arg))
+
+                    if (handler_class_name not in
+                            handlers_with_non_conforming_default_schemas):
+                        handlers_with_non_conforming_default_schemas.append(
+                            handler_class_name)
+
+        error_msg = (
+            'Schema validation for default values failed for handlers: [ %s ].'
+            '\nVisit %s to know, How to write schema for handler args.' % (
+                ', '.join(handlers_with_non_conforming_default_schemas),
+                        self.wiki_page_link))
+
+        self.assertEqual(
+            handlers_with_non_conforming_default_schemas, [], error_msg)
+
+    def test_handlers_with_schema_are_not_in_the_schema_requiring_list(self):
+        """This test checks if a handler contains schema then, handler class
+        name should not be present in SCHEMA_REQUIRING_HANDLER_CLASS_NAMES list.
```

Comment:
I'm a bit confused by this. The name seems to not match the one listed on line 1670.

Maybe you can just say `the list of handlers which do not require schemas`.

Reply:
updated, ptal

Reply:
No, this is still stated wrongly. There is no such list as you mentioned in the docstring.

I've already suggested a rewording of the last part of the docstring in my previous comment. Try to rename the test so that it actually matches what you're testing; work with your mentor if you're not sure how to do this.